apps/web/components/Organization/Slice/TabSection/TabSection.tsx

```diff
@@ -23,37 +29,49 @@ interface SliceProps {
 
 export const TabSectionSlice: React.FC<SliceProps> = ({
   slice,
-  contentColumnProps = {
-    span: ['9/9', '9/9', '9/9', '7/9'],
-    offset: [null, null, null, '1/9'],
-  },
   contentPaddingTop = [0, 4, 6],
 }) => {
+  const [tabSection, setTabSection] = useState(slice)
   const router = useRouter()
+  const { activeLocale } = useI18n()
 
   const selected = useMemo(() => {
-    const index = slice.tabs?.findIndex(
+    const index = tabSection?.tabs?.findIndex(
       (tab) =>
-        tab?.tabTitle && slugify(tab.tabTitle) === router?.query?.selectedTab,
+        tab?.tabTitle && slugify(tab.tabTitle) === router.query?.selectedTab,
     )
     if (index >= 0) {
       return String(index)
     }
     return undefined
-  }, [router.query?.selectedTab, slice.tabs])
+  }, [router.query?.selectedTab, tabSection?.tabs])
+
+  useQuery<GetTabSectionQuery, QueryGetTabSectionArgs>(GET_TAB_SECTION_QUERY, {
+    variables: {
+      input: {
+        id: tabSection?.id,
+        lang: activeLocale,
+      },
+    },
+    onCompleted(data) {
+      if (data?.getTabSection?.tabs?.length) {
+        setTabSection(data.getTabSection as TabSection)
```

Comment:
Is putting the data in a state better than using the output from `useQuery` ? e.g.
```
const { loading, error, data } = useQuery(...
```

Reply:
I did it this way because I'm replacing what's already present after I've gotten data with more depth.
So the site is still fully populated and will only re-render the TabSection once I've gotten the data from contentful